integration/rails_6.0_test.go

```diff
@@ -0,0 +1,161 @@
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+func testRails60(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		imageIDs     map[string]struct{}
+		containerIDs map[string]struct{}
+
+		name       string
+		source     string
+		buildpacks = []string{
+			settings.Buildpacks.MRI.Online,
+			settings.Buildpacks.Bundler.Online,
+			settings.Buildpacks.BundleInstall.Online,
+			settings.Buildpacks.NodeEngine.Online,
+			settings.Buildpacks.Yarn.Online,
+			settings.Buildpacks.YarnInstall.Online,
+			settings.Buildpacks.RailsAssets.Online,
+			settings.Buildpacks.Puma.Online,
+		}
+	)
+
+	it.Before(func() {
+		imageIDs = make(map[string]struct{})
+		containerIDs = make(map[string]struct{})
+
+		var err error
+		name, err = occam.RandomName()
+		Expect(err).NotTo(HaveOccurred())
+
+		source, err = occam.Source(filepath.Join("testdata", "6.0"))
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	it.After(func() {
+		for id := range containerIDs {
+			Expect(settings.Docker.Container.Remove.Execute(id)).To(Succeed())
+		}
+
+		for id := range imageIDs {
+			Expect(settings.Docker.Image.Remove.Execute(id)).To(Succeed())
+		}
+
+		Expect(settings.Docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+		Expect(os.RemoveAll(source)).To(Succeed())
+	})
+
+	it("creates a working OCI image with compiled rails assets", func() {
+		image, logs, err := settings.Pack.WithNoColor().Build.
+			WithBuildpacks(buildpacks...).
+			WithPullPolicy("never").
+			Execute(name, source)
+		Expect(err).NotTo(HaveOccurred(), logs.String())
+
+		imageIDs[image.ID] = struct{}{}
+
+		container, err := settings.Docker.Container.Run.
+			WithEnv(map[string]string{
+				"PORT":            "8080",
+				"SECRET_KEY_BASE": "some-secret",
+			}).
+			WithPublish("8080").
+			WithPublishAll().
+			Execute(image.ID)
+		Expect(err).NotTo(HaveOccurred())
+
+		containerIDs[container.ID] = struct{}{}
+
+		Eventually(container).Should(BeAvailable())
+
+		response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort("8080")))
+		Expect(err).NotTo(HaveOccurred())
+		defer response.Body.Close()
+
+		Expect(response.StatusCode).To(Equal(http.StatusOK), func() string {
+			dump, _ := httputil.DumpResponse(response, true)
+			logs, _ := settings.Docker.Container.Logs.Execute(container.ID)
+			return fmt.Sprintf("reponse:\n%s\n\nlogs:\n%s\n", dump, logs)
+		})
+
+		content, err := ioutil.ReadAll(response.Body)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(content)).To(ContainSubstring("Hello World"))
+
+		Expect(logs).To(ContainLines(
+			MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
+			"  Executing build process",
+			"    Running 'bundle exec rails assets:precompile assets:clean'",
```

Comment:
Do you think we should also check the logs for the `yarn install` process since this is an important distinction in the Rails 6.0 flow?

Reply:
I think the fact that the app detects and builds is sufficient to say that the `yarn install` process was successful. I don't think we need to also assert that we see something in the logs. We'd just be unnecessarily coupling this test to the output of that buildpack.

Reply:
That's valid